Kubelet unit tests broken in master

```
--- FAIL: TestUpdateNewNodeStatus (0.00s)
        kubelet_test.go:3295: unexpected error: update node status exceeds retry count
        kubelet_test.go:3298: unexpected actions: [{get-node testnode} {get-node testnode} {get-node testnode} {get-node testnode} {get-node testnode}]
I0519 16:55:50.176533   23160 plugins.go:56] Registering credential provider: .dockercfg
E0519 16:55:50.176822   23160 kubelet.go:1775] Error updating node status, will retry: can't get ip address of node 127.0.0.1: lookup 127.0.0.1: invalid domain name
E0519 16:55:50.176960   23160 kubelet.go:1775] Error updating node status, will retry: can't get ip address of node 127.0.0.1: lookup 127.0.0.1: invalid domain name
E0519 16:55:50.177090   23160 kubelet.go:1775] Error updating node status, will retry: can't get ip address of node 127.0.0.1: lookup 127.0.0.1: invalid domain name
E0519 16:55:50.177213   23160 kubelet.go:1775] Error updating node status, will retry: can't get ip address of node 127.0.0.1: lookup 127.0.0.1: invalid domain name
E0519 16:55:50.177358   23160 kubelet.go:1775] Error updating node status, will retry: can't get ip address of node 127.0.0.1: lookup 127.0.0.1: invalid domain name
--- FAIL: TestUpdateExistingNodeStatus (0.00s)
        kubelet_test.go:3392: unexpected error: update node status exceeds retry count
        kubelet_test.go:3395: unexpected actions: [{get-node testnode} {get-node testnode} {get-node testnode} {get-node testnode} {get-node testnode}]
        kubelet_test.go:3399: unexpected object type
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
        panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xb code=0x1 addr=0x0 pc=0x4d56a7]

goroutine 94 [running]:
testing.func·006()
        /usr/lib/golang/src/testing/testing.go:441 +0x181
github.com/GoogleCloudPlatform/kubernetes/pkg/kubelet.TestUpdateExistingNodeStatus(0xc20806be60)
        /home/pmorie/code/kubernetes/_output/local/go/src/github.com/GoogleCloudPlatform/kubernetes/pkg/kubelet/kubelet_test.go:3402 +0x2157
testing.tRunner(0xc20806be60, 0x14db238)
        /usr/lib/golang/src/testing/testing.go:447 +0xbf
created by testing.RunTests
        /usr/lib/golang/src/testing/testing.go:555 +0xa8b

goroutine 1 [chan receive]:
testing.RunTests(0xf2b680, 0x14dac20, 0x79, 0x79, 0x0)
        /usr/lib/golang/src/testing/testing.go:556 +0xad6
testing.(*M).Run(0xc208049ae0, 0x14fe640)
        /usr/lib/golang/src/testing/testing.go:485 +0x6c
main.main()
        github.com/GoogleCloudPlatform/kubernetes/pkg/kubelet/_test/_testmain.go:292 +0x1d5

goroutine 5 [chan receive]:
github.com/golang/glog.(*loggingT).flushDaemon(0x14f60e0)
        /home/pmorie/code/kubernetes/Godeps/_workspace/src/github.com/golang/glog/glog.go:879 +0x78
created by github.com/golang/glog.init·1
        /home/pmorie/code/kubernetes/Godeps/_workspace/src/github.com/golang/glog/glog.go:410 +0x2a7

goroutine 17 [syscall, locked to thread]:
runtime.goexit()
        /usr/lib/golang/src/runtime/asm_amd64.s:2232 +0x1
FAIL    github.com/GoogleCloudPlatform/kubernetes/pkg/kubelet   0.046s
```

@zmerlynn @vmarmol

seeing a similar failure

```
W0519 21:43:05.753721    3904 docker.go:247] found a container with the "k8s" prefix, but too few fields (2): "k8s_unidentified"
I0519 21:43:05.754128    3904 container_gc.go:126] Removing unidentified dead container "/k8s_unidentified" with ID "2876"
I0519 21:43:05.755215    3904 image_manager.go:254] [ImageManager]: Removing image "image-0" to free 1024 bytes
I0519 21:43:05.755375    3904 image_manager.go:254] [ImageManager]: Removing image "image-0" to free 1024 bytes
I0519 21:43:05.755526    3904 image_manager.go:254] [ImageManager]: Removing image "image-0" to free 1024 bytes
I0519 21:43:05.755642    3904 image_manager.go:254] [ImageManager]: Removing image "image-0" to free 1024 bytes
I0519 21:43:05.755796    3904 image_manager.go:202] [ImageManager]: Disk usage on "" () is at 95% which is over the high threshold (90%). Trying to free 150 bytes
I0519 21:43:05.755850    3904 image_manager.go:254] [ImageManager]: Removing image "image-0" to free 450 bytes
I0519 21:43:05.755904    3904 image_manager.go:202] [ImageManager]: Disk usage on "" () is at 95% which is over the high threshold (90%). Trying to free 150 bytes
I0519 21:43:05.755981    3904 image_manager.go:254] [ImageManager]: Removing image "image-0" to free 50 bytes
I0519 21:43:05.756235    3904 plugins.go:56] Registering credential provider: .dockercfg
I0519 21:43:05.756590    3904 plugins.go:56] Registering credential provider: .dockercfg
W0519 21:43:05.756746    3904 kubelet.go:521] Data dir for pod "bothpod" exists in both old and new form, using new
W0519 21:43:05.756958    3904 kubelet.go:572] Data dir for pod "newpod", container "bothctr" exists in both old and new form, using new
I0519 21:43:05.757230    3904 plugins.go:56] Registering credential provider: .dockercfg
E0519 21:43:05.757470    3904 kubelet.go:1523] error getting node: node testnode not found
E0519 21:43:05.757906    3904 kubelet.go:2097] Cannot get host IP: cannot get node: node testnode not found
E0519 21:43:05.758089    3904 kubelet.go:2097] Cannot get host IP: cannot get node: node testnode not found
I0519 21:43:05.758534    3904 plugins.go:56] Registering credential provider: .dockercfg
E0519 21:43:05.758630    3904 kubelet.go:1523] error getting node: node testnode not found
E0519 21:43:05.758925    3904 kubelet.go:2097] Cannot get host IP: cannot get node: node testnode not found
E0519 21:43:05.759962    3904 kubelet.go:2097] Cannot get host IP: cannot get node: node testnode not found
I0519 21:43:05.760435    3904 plugins.go:56] Registering credential provider: .dockercfg
E0519 21:43:05.760528    3904 kubelet.go:1523] error getting node: node testnode not found
E0519 21:43:05.760758    3904 kubelet.go:2097] Cannot get host IP: cannot get node: node testnode not found
E0519 21:43:05.761656    3904 kubelet.go:2097] Cannot get host IP: cannot get node: node testnode not found
I0519 21:43:05.762088    3904 plugins.go:56] Registering credential provider: .dockercfg
E0519 21:43:05.762170    3904 kubelet.go:1523] error getting node: node testnode not found
E0519 21:43:05.762380    3904 kubelet.go:2097] Cannot get host IP: cannot get node: node testnode not found
E0519 21:43:05.764953    3904 kubelet.go:2097] Cannot get host IP: cannot get node: node testnode not found
I0519 21:43:05.765590    3904 plugins.go:56] Registering credential provider: .dockercfg
E0519 21:43:05.765734    3904 kubelet.go:1523] error getting node: node testnode not found
E0519 21:43:05.766249    3904 kubelet.go:2097] Cannot get host IP: cannot get node: node testnode not found
E0519 21:43:05.766519    3904 kubelet.go:2097] Cannot get host IP: cannot get node: node testnode not found
I0519 21:43:05.767080    3904 plugins.go:56] Registering credential provider: .dockercfg
I0519 21:43:05.767132    3904 plugins.go:56] Registering credential provider: .dockercfg
E0519 21:43:05.767230    3904 kubelet.go:1523] error getting node: node testnode not found
E0519 21:43:05.767519    3904 kubelet.go:2097] Cannot get host IP: cannot get node: node testnode not found
E0519 21:43:05.767971    3904 kubelet.go:2097] Cannot get host IP: cannot get node: node testnode not found
I0519 21:43:05.768335    3904 plugins.go:56] Registering credential provider: .dockercfg
E0519 21:43:05.768442    3904 kubelet.go:1523] error getting node: node testnode not found
E0519 21:43:05.768704    3904 kubelet.go:2097] Cannot get host IP: cannot get node: node testnode not found
W0519 21:43:05.768785    3904 manager.go:1187] No ref for pod '1234'
E0519 21:43:05.769438    3904 kubelet.go:2097] Cannot get host IP: cannot get node: node testnode not found
E0519 21:43:05.769644    3904 kubelet.go:2097] Cannot get host IP: cannot get node: node testnode not found
E0519 21:43:05.769777    3904 kubelet.go:2097] Cannot get host IP: cannot get node: node testnode not found
I0519 21:43:05.770232    3904 plugins.go:56] Registering credential provider: .dockercfg
E0519 21:43:05.770278    3904 kubelet.go:1523] error getting node: node testnode not found
E0519 21:43:05.770364    3904 kubelet.go:1523] error getting node: node testnode not found
W0519 21:43:05.770447    3904 manager.go:1187] No ref for pod '1234'
W0519 21:43:05.770471    3904 manager.go:1187] No ref for pod '9876'
I0519 21:43:05.770723    3904 plugins.go:56] Registering credential provider: .dockercfg
E0519 21:43:05.770769    3904 kubelet.go:1523] error getting node: node testnode not found
W0519 21:43:05.770834    3904 manager.go:1187] No ref for pod '1234'
W0519 21:43:05.770855    3904 manager.go:1187] No ref for pod '9876'
I0519 21:43:05.771126    3904 plugins.go:56] Registering credential provider: .dockercfg
E0519 21:43:05.771225    3904 kubelet.go:1523] error getting node: node testnode not found
E0519 21:43:05.771503    3904 kubelet.go:2097] Cannot get host IP: cannot get node: node testnode not found
W0519 21:43:05.771590    3904 manager.go:1187] No ref for pod '1234'
E0519 21:43:05.771657    3904 kubelet.go:2097] Cannot get host IP: cannot get node: node testnode not found
I0519 21:43:05.772008    3904 plugins.go:56] Registering credential provider: .dockercfg
E0519 21:43:05.772099    3904 kubelet.go:1523] error getting node: node testnode not found
E0519 21:43:05.774180    3904 kubelet.go:2097] Cannot get host IP: cannot get node: node testnode not found
I0519 21:43:05.774300    3904 manager.go:1392] pod "foo_new" container "bar" hash changed (4660 vs 2312944361), it will be killed and re-created.
W0519 21:43:05.774353    3904 manager.go:1187] No ref for pod '1234'
E0519 21:43:05.774574    3904 kubelet.go:2097] Cannot get host IP: cannot get node: node testnode not found
I0519 21:43:05.775132    3904 plugins.go:56] Registering credential provider: .dockercfg
E0519 21:43:05.775247    3904 kubelet.go:1523] error getting node: node testnode not found
E0519 21:43:05.775599    3904 kubelet.go:2097] Cannot get host IP: cannot get node: node testnode not found
W0519 21:43:05.775713    3904 prober.go:195] Failed to find probe builder for container: {bar  [] []  [] [] {map[] map[]} [] 0xc20823f710 <nil> <nil>   <nil>}
W0519 21:43:05.776201    3904 prober.go:195] Failed to find probe builder for container: {bar  [] []  [] [] {map[] map[]} [] 0xc20823f710 <nil> <nil>   <nil>}
W0519 21:43:05.776229    3904 prober.go:195] Failed to find probe builder for container: {bar  [] []  [] [] {map[] map[]} [] 0xc20823f710 <nil> <nil>   <nil>}
I0519 21:43:05.776253    3904 manager.go:1409] pod "foo_new" container "bar" is unhealthy (probe result: unknown), it will be killed and re-created.
W0519 21:43:05.776296    3904 manager.go:1187] No ref for pod '1234'
E0519 21:43:05.776539    3904 kubelet.go:2097] Cannot get host IP: cannot get node: node testnode not found
I0519 21:43:05.777107    3904 plugins.go:56] Registering credential provider: .dockercfg
I0519 21:43:05.778569    3904 plugins.go:56] Registering credential provider: .dockercfg
I0519 21:43:05.779262    3904 plugins.go:56] Registering credential provider: .dockercfg
I0519 21:43:05.780126    3904 plugins.go:56] Registering credential provider: .dockercfg
I0519 21:43:05.780465    3904 plugins.go:56] Registering credential provider: .dockercfg
I0519 21:43:05.780596    3904 plugins.go:56] Registering credential provider: .dockercfg
I0519 21:43:05.780750    3904 plugins.go:56] Registering credential provider: .dockercfg
I0519 21:43:05.782402    3904 plugins.go:56] Registering credential provider: .dockercfg
I0519 21:43:05.782803    3904 plugins.go:56] Registering credential provider: .dockercfg
I0519 21:43:05.783351    3904 plugins.go:56] Registering credential provider: .dockercfg
I0519 21:43:05.783559    3904 plugins.go:56] Registering credential provider: .dockercfg
I0519 21:43:05.783590    3904 plugins.go:56] Registering credential provider: .dockercfg
E0519 21:43:05.783674    3904 kubelet.go:1523] error getting node: node testnode not found
E0519 21:43:05.783972    3904 kubelet.go:2097] Cannot get host IP: cannot get node: node testnode not found
E0519 21:43:05.784347    3904 manager.go:1537] Error running pod "foo_new" container "bar": failed to call event handler: test error
E0519 21:43:05.784410    3904 kubelet.go:2097] Cannot get host IP: cannot get node: node testnode not found
I0519 21:43:05.784768    3904 plugins.go:56] Registering credential provider: .dockercfg
E0519 21:43:05.784838    3904 kubelet.go:1523] error getting node: node testnode not found
E0519 21:43:05.785089    3904 kubelet.go:2097] Cannot get host IP: cannot get node: node testnode not found
E0519 21:43:05.786605    3904 kubelet.go:2097] Cannot get host IP: cannot get node: node testnode not found
I0519 21:43:05.787287    3904 plugins.go:56] Registering credential provider: .dockercfg
I0519 21:43:05.787679    3904 plugins.go:56] Registering credential provider: .dockercfg
I0519 21:43:05.787871    3904 plugins.go:56] Registering credential provider: .dockercfg
I0519 21:43:05.789782    3904 plugins.go:56] Registering credential provider: .dockercfg
I0519 21:43:05.790223    3904 plugins.go:56] Registering credential provider: .dockercfg
I0519 21:43:05.791024    3904 plugins.go:56] Registering credential provider: .dockercfg
I0519 21:43:05.791597    3904 plugins.go:56] Registering credential provider: .dockercfg
I0519 21:43:05.791885    3904 plugins.go:56] Registering credential provider: .dockercfg
I0519 21:43:05.792179    3904 plugins.go:56] Registering credential provider: .dockercfg
I0519 21:43:05.792436    3904 plugins.go:56] Registering credential provider: .dockercfg
I0519 21:43:05.792705    3904 plugins.go:56] Registering credential provider: .dockercfg
I0519 21:43:05.792917    3904 plugins.go:56] Registering credential provider: .dockercfg
E0519 21:43:05.793042    3904 kubelet.go:1449] Pod "_": HostPort is already allocated, ignoring: [[0].port: duplicate value '81']
I0519 21:43:05.793434    3904 plugins.go:56] Registering credential provider: .dockercfg
E0519 21:43:05.793522    3904 kubelet.go:1449] Pod "newpod_foo": HostPort is already allocated, ignoring: [[0].port: duplicate value '80']
E0519 21:43:05.793640    3904 kubelet.go:1523] error getting node: node testnode not found
I0519 21:43:05.795469    3904 plugins.go:56] Registering credential provider: .dockercfg
I0519 21:43:05.795971    3904 plugins.go:56] Registering credential provider: .dockercfg
E0519 21:43:05.796057    3904 kubelet.go:1523] error getting node: node testnode not found
I0519 21:43:05.796608    3904 plugins.go:56] Registering credential provider: .dockercfg
E0519 21:43:05.796673    3904 kubelet.go:1449] Pod "pod2_": HostPort is already allocated, ignoring: [[0].port: duplicate value '80']
E0519 21:43:05.796840    3904 kubelet.go:1523] error getting node: node testnode not found
E0519 21:43:05.796895    3904 kubelet.go:1523] error getting node: node testnode not found
I0519 21:43:05.797164    3904 plugins.go:56] Registering credential provider: .dockercfg
I0519 21:43:05.797344    3904 plugins.go:56] Registering credential provider: .dockercfg
I0519 21:43:05.797520    3904 plugins.go:56] Registering credential provider: .dockercfg
E0519 21:43:05.798431    3904 kubelet.go:1775] Error updating node status, will retry: can't get ip address of node 127.0.0.1: lookup 127.0.0.1: invalid domain name
E0519 21:43:05.798605    3904 kubelet.go:1775] Error updating node status, will retry: can't get ip address of node 127.0.0.1: lookup 127.0.0.1: invalid domain name
E0519 21:43:05.798755    3904 kubelet.go:1775] Error updating node status, will retry: can't get ip address of node 127.0.0.1: lookup 127.0.0.1: invalid domain name
E0519 21:43:05.798905    3904 kubelet.go:1775] Error updating node status, will retry: can't get ip address of node 127.0.0.1: lookup 127.0.0.1: invalid domain name
E0519 21:43:05.799031    3904 kubelet.go:1775] Error updating node status, will retry: can't get ip address of node 127.0.0.1: lookup 127.0.0.1: invalid domain name
--- FAIL: TestUpdateNewNodeStatus (0.00 seconds)
	kubelet_test.go:3295: unexpected error: update node status exceeds retry count
	kubelet_test.go:3298: unexpected actions: [{get-node testnode} {get-node testnode} {get-node testnode} {get-node testnode} {get-node testnode}]
I0519 21:43:05.799350    3904 plugins.go:56] Registering credential provider: .dockercfg
E0519 21:43:05.800006    3904 kubelet.go:1775] Error updating node status, will retry: can't get ip address of node 127.0.0.1: lookup 127.0.0.1: invalid domain name
E0519 21:43:05.800273    3904 kubelet.go:1775] Error updating node status, will retry: can't get ip address of node 127.0.0.1: lookup 127.0.0.1: invalid domain name
E0519 21:43:05.800537    3904 kubelet.go:1775] Error updating node status, will retry: can't get ip address of node 127.0.0.1: lookup 127.0.0.1: invalid domain name
E0519 21:43:05.800873    3904 kubelet.go:1775] Error updating node status, will retry: can't get ip address of node 127.0.0.1: lookup 127.0.0.1: invalid domain name
E0519 21:43:05.801130    3904 kubelet.go:1775] Error updating node status, will retry: can't get ip address of node 127.0.0.1: lookup 127.0.0.1: invalid domain name
--- FAIL: TestUpdateExistingNodeStatus (0.00 seconds)
	kubelet_test.go:3392: unexpected error: update node status exceeds retry count
	kubelet_test.go:3395: unexpected actions: [{get-node testnode} {get-node testnode} {get-node testnode} {get-node testnode} {get-node testnode}]
	kubelet_test.go:3399: unexpected object type
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xb code=0x1 addr=0x0 pc=0x4a6ff0]

goroutine 99 [running]:
runtime.panic(0xc65740, 0x1516cb3)
	/usr/lib/golang/src/pkg/runtime/panic.c:279 +0xf5
testing.func·006()
	/usr/lib/golang/src/pkg/testing/testing.go:416 +0x176
runtime.panic(0xc65740, 0x1516cb3)
	/usr/lib/golang/src/pkg/runtime/panic.c:248 +0x18d
github.com/GoogleCloudPlatform/kubernetes/pkg/kubelet.TestUpdateExistingNodeStatus(0xc2080702d0)
	/home/git/origin/kubernetes/_output/local/go/src/github.com/GoogleCloudPlatform/kubernetes/pkg/kubelet/kubelet_test.go:3402 +0x1d10
testing.tRunner(0xc2080702d0, 0x151ea58)
	/usr/lib/golang/src/pkg/testing/testing.go:422 +0x8b
created by testing.RunTests
	/usr/lib/golang/src/pkg/testing/testing.go:504 +0x8db

goroutine 16 [chan receive]:
testing.RunTests(0xf0ccd0, 0x151e440, 0x79, 0x79, 0x0)
	/usr/lib/golang/src/pkg/testing/testing.go:505 +0x923
testing.Main(0xf0ccd0, 0x151e440, 0x79, 0x79, 0x153a100, 0x0, 0x0, 0x153a100, 0x0, 0x0)
	/usr/lib/golang/src/pkg/testing/testing.go:435 +0x84
main.main()
	github.com/GoogleCloudPlatform/kubernetes/pkg/kubelet/_test/_testmain.go:287 +0x9c

goroutine 19 [finalizer wait]:
runtime.park(0x4144d0, 0x1532028, 0x151bfe9)
	/usr/lib/golang/src/pkg/runtime/proc.c:1369 +0x89
runtime.parkunlock(0x1532028, 0x151bfe9)
	/usr/lib/golang/src/pkg/runtime/proc.c:1385 +0x3b
runfinq()
	/usr/lib/golang/src/pkg/runtime/mgc0.c:2644 +0xcf
runtime.goexit()
	/usr/lib/golang/src/pkg/runtime/proc.c:1445

goroutine 20 [chan receive]:
github.com/golang/glog.(*loggingT).flushDaemon(0x1536020)
	/home/git/origin/kubernetes/Godeps/_workspace/src/github.com/golang/glog/glog.go:879 +0x75
created by github.com/golang/glog.init·1
	/home/git/origin/kubernetes/Godeps/_workspace/src/github.com/golang/glog/glog.go:410 +0x2b2

goroutine 17 [syscall]:
runtime.goexit()
	/usr/lib/golang/src/pkg/runtime/proc.c:1445
FAIL	github.com/GoogleCloudPlatform/kubernetes/pkg/kubelet	0.077s

```

--- a/pkg/kubelet/kubelet_test.go
+++ b/pkg/kubelet/kubelet_test.go
@@ -69,7 +69,7 @@ type TestKubelet struct {
 	fakeMirrorClient *fakeMirrorClient
 }
 
-const testKubeletHostname = "testnode"
+const testKubeletHostname = "127.0.0.1"
 
 func newTestKubelet(t *testing.T) *TestKubelet {
 	fakeDocker := &dockertools.FakeDockerClient{Errors: make(map[string]error), RemovedImages: util.StringSet{}}
@@ -81,7 +81,7 @@ func newTestKubelet(t *testing.T) *TestKubelet {
 	kubelet.kubeClient = fakeKubeClient
 	kubelet.os = kubecontainer.FakeOS{}
 
-	kubelet.hostname = "testnode"
+	kubelet.hostname = testKubeletHostname
 	kubelet.runtimeUpThreshold = maxWaitForContainerRuntime
 	kubelet.networkPlugin, _ = network.InitNetworkPlugin([]network.NetworkPlugin{}, "", network.NewFakeHost(nil))
 	if tempDir, err := ioutil.TempDir("/tmp", "kubelet_test."); err != nil {
@@ -3026,7 +3026,7 @@ func TestHandleNodeSelector(t *testing.T) {
 	testKubelet := newTestKubelet(t)
 	kl := testKubelet.kubelet
 	kl.nodeLister = testNodeLister{nodes: []api.Node{
-		{ObjectMeta: api.ObjectMeta{Name: "testnode", Labels: map[string]string{"key": "B"}}},
+		{ObjectMeta: api.ObjectMeta{Name: testKubeletHostname, Labels: map[string]string{"key": "B"}}},
 	}}
 	testKubelet.fakeCadvisor.On("MachineInfo").Return(&cadvisorApi.MachineInfo{}, nil)
 	testKubelet.fakeCadvisor.On("DockerImagesFsInfo").Return(cadvisorApiv2.FsInfo{}, nil)
@@ -3241,7 +3241,7 @@ func TestUpdateNewNodeStatus(t *testing.T) {
 	kubelet := testKubelet.kubelet
 	kubeClient := testKubelet.fakeKubeClient
 	kubeClient.ReactFn = testclient.NewSimpleFake(&api.NodeList{Items: []api.Node{
-		{ObjectMeta: api.ObjectMeta{Name: "127.0.0.1"}},
+		{ObjectMeta: api.ObjectMeta{Name: testKubeletHostname}},
 	}}).ReactFn
 	machineInfo := &cadvisorApi.MachineInfo{
 		MachineID:      "123",
@@ -3259,7 +3259,7 @@ func TestUpdateNewNodeStatus(t *testing.T) {
 	}
 	mockCadvisor.On("VersionInfo").Return(versionInfo, nil)
 	expectedNode := &api.Node{
-		ObjectMeta: api.ObjectMeta{Name: "127.0.0.1"},
+		ObjectMeta: api.ObjectMeta{Name: testKubeletHostname},
 		Spec:       api.NodeSpec{},
 		Status: api.NodeStatus{
 			Conditions: []api.NodeCondition{
@@ -3320,7 +3320,7 @@ func TestUpdateExistingNodeStatus(t *testing.T) {
 	kubeClient := testKubelet.fakeKubeClient
 	kubeClient.ReactFn = testclient.NewSimpleFake(&api.NodeList{Items: []api.Node{
 		{
-			ObjectMeta: api.ObjectMeta{Name: "127.0.0.1"},
+			ObjectMeta: api.ObjectMeta{Name: testKubeletHostname},
 			Spec:       api.NodeSpec{},
 			Status: api.NodeStatus{
 				Conditions: []api.NodeCondition{
@@ -3356,7 +3356,7 @@ func TestUpdateExistingNodeStatus(t *testing.T) {
 	}
 	mockCadvisor.On("VersionInfo").Return(versionInfo, nil)
 	expectedNode := &api.Node{
-		ObjectMeta: api.ObjectMeta{Name: "127.0.0.1"},
+		ObjectMeta: api.ObjectMeta{Name: testKubeletHostname},
 		Spec:       api.NodeSpec{},
 		Status: api.NodeStatus{
 			Conditions: []api.NodeCondition{
@@ -3423,7 +3423,7 @@ func TestUpdateNodeStatusWithoutContainerRuntime(t *testing.T) {
 	fakeDocker.VersionInfo = []string{}
 
 	kubeClient.ReactFn = testclient.NewSimpleFake(&api.NodeList{Items: []api.Node{
-		{ObjectMeta: api.ObjectMeta{Name: "127.0.0.1"}},
+		{ObjectMeta: api.ObjectMeta{Name: testKubeletHostname}},
 	}}).ReactFn
 	mockCadvisor := testKubelet.fakeCadvisor
 	machineInfo := &cadvisorApi.MachineInfo{
@@ -3442,7 +3442,7 @@ func TestUpdateNodeStatusWithoutContainerRuntime(t *testing.T) {
 	mockCadvisor.On("VersionInfo").Return(versionInfo, nil)
 
 	expectedNode := &api.Node{
-		ObjectMeta: api.ObjectMeta{Name: "127.0.0.1"},
+		ObjectMeta: api.ObjectMeta{Name: testKubeletHostname},
 		Spec:       api.NodeSpec{},
 		Status: api.NodeStatus{
 			Conditions: []api.NodeCondition{
@@ -4410,7 +4410,6 @@ func TestFilterOutTerminatedPods(t *testing.T) {
 func TestRegisterExistingNodeWithApiserver(t *testing.T) {
 	testKubelet := newTestKubelet(t)
 	kubelet := testKubelet.kubelet
-	kubelet.hostname = "127.0.0.1"
 	kubeClient := testKubelet.fakeKubeClient
 	kubeClient.ReactFn = func(action testclient.FakeAction) (runtime.Object, error) {
 		segments := strings.Split(action.Action, "-")
@@ -4427,8 +4426,8 @@ func TestRegisterExistingNodeWithApiserver(t *testing.T) {
 		case "get":
 			// Return an existing (matching) node on get.
 			return &api.Node{
-				ObjectMeta: api.ObjectMeta{Name: "127.0.0.1"},
-				Spec:       api.NodeSpec{ExternalID: "127.0.0.1"},
+				ObjectMeta: api.ObjectMeta{Name: testKubeletHostname},
+				Spec:       api.NodeSpec{ExternalID: testKubeletHostname},
 			}, nil
 		default:
 			return nil, fmt.Errorf("no reaction implemented for %s", action.Action)

